internal/common/customplanmodifier/non_updatable.go

```diff
@@ -0,0 +1,72 @@
+package customplanmodifier
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	planmodifier "github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func NonUpdatableAttributePlanModifier(attribute string) planmodifier.String {
```

Comment:
in the name would mention it is string specific, eventually we might have similar implementation for other types.